app/scripts/metamask-controller.js

```diff
@@ -532,6 +535,19 @@ export default class MetamaskController extends EventEmitter {
       trackMetaMetricsEvent: (...args) =>
         this.metaMetricsController.trackEvent(...args),
     });
+
+    this.multichainSubscriptionManager = new MultichainSubscriptionManager({
+      getNetworkClientById: this.networkController.getNetworkClientById.bind(
+        this.networkController,
+      ),
+      findNetworkClientIdByChainId:
+        this.networkController.findNetworkClientIdByChainId.bind(
+          this.networkController,
+        ),
+    });
+
+    this.multichainMiddlewareManager = createMultichainMiddlewareManager();
```

Comment:
i think these have to be aware of connection/origin or must be instantiated per connection/origin because it seems in the current implementation it may be possible for dapps to receive eth_subscriptions intended for a different dapp.

Reply:
💯 ➕